client/opentracing/src/test/java/org/hawkular/apm/client/opentracing/APMTracerReferenceTest.java

```diff
@@ -0,0 +1,457 @@
+
+        assertEquals(1, testTraceReporter.getTraces().size());
+
+        Trace trace = testTraceReporter.getTraces().get(0);
+        assertEquals(1, trace.getNodes().size());
+        assertEquals(Consumer.class, trace.getNodes().get(0).getClass());
+        assertEquals(((Consumer) trace.getNodes().get(0)).getCorrelationIds().get(0),
+                new CorrelationIdentifier(Scope.Interaction, TEST_APM_ID1));
+        assertEquals(0, ((Consumer) trace.getNodes().get(0)).getNodes().size());
+    }
+
+    @Test
+    public void testSingleChildOfSpan() {
+        TestTraceReporter reporter = new TestTraceReporter();
+        Tracer tracer = new APMTracer(reporter);
+
+        Span parentSpan = tracer.buildSpan("ParentSpan")
+                .start();
+
+        Span childSpan = tracer.buildSpan("ChildSpan")
+                .asChildOf(parentSpan)
+                .start();
+        childSpan.finish();
+
+        parentSpan.finish();
+
+        assertEquals(1, reporter.getTraces().size());
+
+        Trace trace = reporter.getTraces().get(0);
+        assertEquals(1, trace.getNodes().size());
+        assertEquals(Component.class, trace.getNodes().get(0).getClass());
+
+        Component parentComponent = (Component) trace.getNodes().get(0);
+
+        assertTrue(parentComponent.getCorrelationIds().isEmpty());
+        assertEquals(1, parentComponent.getNodes().size());
+        assertEquals(Component.class, parentComponent.getNodes().get(0).getClass());
+
+        Component childComponent = (Component) parentComponent.getNodes().get(0);
+
+        assertTrue(childComponent.getCorrelationIds().isEmpty());
+        assertEquals(0, childComponent.getNodes().size());
+    }
+
+    @Test
+    public void testSingleChildOfSpanUsingContext() {
+        TestTraceReporter reporter = new TestTraceReporter();
+        Tracer tracer = new APMTracer(reporter);
+
+        Span parentSpan = tracer.buildSpan("ParentSpan")
+                .start();
+
+        Span childSpan = tracer.buildSpan("ChildSpan")
+                .asChildOf(parentSpan.context())
+                .start();
+        childSpan.finish();
+
+        parentSpan.finish();
+
+        assertEquals(1, reporter.getTraces().size());
+
+        Trace trace = reporter.getTraces().get(0);
+        assertEquals(1, trace.getNodes().size());
+        assertEquals(Component.class, trace.getNodes().get(0).getClass());
+
+        Component parentComponent = (Component) trace.getNodes().get(0);
+
+        assertTrue(parentComponent.getCorrelationIds().isEmpty());
+        assertEquals(1, parentComponent.getNodes().size());
+        assertEquals(Component.class, parentComponent.getNodes().get(0).getClass());
+
+        Component childComponent = (Component) parentComponent.getNodes().get(0);
+
+        assertTrue(childComponent.getCorrelationIds().isEmpty());
+        assertEquals(0, childComponent.getNodes().size());
+    }
+
+    @Test
+    public void testSingleFollowsFromRef() {
```

Comment:
It would be good to set http.url tag on parent span and test  operation, URI and endpoint on followsFrom Consumer.